src/ansys/pyensight/core/utils/adr.py

```diff
@@ -0,0 +1,67 @@
+from types import ModuleType
+from typing import Union
+
+try:
+    import ensight
+except ImportError:
+    from ansys.api.pyensight import ensight_api
+
+
+class Adr:
+    """Provides the ``ensight.utils.adr`` interface.
+
+    The methods in this class implement simplified interfaces connect EnSight
+    to an existing ADR report.
+    With connect_to_existing_adr_report() the connection is done, while with
+    generate_adr_report() the report is generated.
+
+    Parameters
+    ----------
+    interface :
+        Entity that provides the ``ensight`` namespace. In the case of
+        EnSight Python, the ``ensight`` module is passed. In the case
+        of PyEnSight, ``Session.ensight`` is passed.
+    """
+
+    def __init__(self, interface: Union["ensight_api.ensight", "ensight"]):
+        self._ensight = interface
+        self._adr_report_connected = False
+
+    def connect_to_existing_adr_report(self, url: str, username: str, password: str) -> None:
+        """Connect to an existing ADR report
+
+        Parameters
+        ----------
+        url: string
+            the url of the ADR report
+        username: string
+            the username to connect to the ADR report
+        password: string
+            the password to connect to the ADR report
+        """
+        if isinstance(self._ensight, ModuleType):
+            self._ensight.core.nexus.ReportServer.get_server().set_URL(url)
+            self._ensight.core.nexus.ReportServer.get_server().set_username(username)
+            self._ensight.core.nexus.ReportServer.get_server().set_password(password)
+        else:
+            self._ensight._session.cmd(
+                f"ensight.core.nexus.ReportServer.get_server().set_URL('{url}')"
+            )
+            self._ensight._session.cmd(
+                f"ensight.core.nexus.ReportServer.get_server().set_username('{username}')"
+            )
+            self._ensight._session.cmd(
+                f"ensight.core.nexus.ReportServer.get_server().set_password('{password}')"
+            )
```

Comment:
@mariostieriansys I'd add here a validate the report just to make sure url, username and password are valid and you're not connecting to a non-existing server before setting adr_report_connected to True. Also I might check if you're already connected to something before re-doing the connection. But it's not strictly needed.